src/lib/util/util.ts

```diff
@@ -547,6 +547,30 @@ export function enumerable(value: boolean) {
 	};
 }
 
+export function ApplyOptions<T extends PieceOptions>(options: T): Function {
+
+	return (target: Constructor<Piece>): Constructor<Piece> => class extends target {
+
+		public constructor(store: Store<string, Piece, typeof Piece>, file: string[], directory: string) {
+			super(store, file, directory, options);
+		}
+
+	};
+
+}
+
+export function CreateResolver(name: string, fn: ArgResolverCustomMethod): Function {
+	return (target: Constructor<Command>) => class extends target {
+
+		public constructor(store: CommandStore, file: string[], directory: string, options: CommandOptions) {
+			super(store, file, directory, options);
+
+			this.createCustomResolver(name, fn);
+		}
+
+	};
+}
```

Comment:
Those should be moved to https://github.com/kyranet/klasa-decorators
And as a matter of a fact, @Favna is already working on it: https://github.com/kyranet/klasa-decorators/pull/19

Reply:
Also, I'm not very fond of having to... creating a class every time those are used, we should investigate a way to do something like `@Batch` which implements an array as a non-enumerable static symbol-keyed property that is called when the constructor is, likewise how Klasa's plugin system works.

That needs some investigation, I know it's completely possible and to be honest, it should be done over having to create multiple classes dynamically, which take a lot of memory and makes inheritance checks significantly slower.